quadriga/src/main/java/edu/asu/diging/quadriga/web/service/impl/GraphCreationServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+
+            node.setUri(sourceURI);
+            CachedConcept conceptCache = conceptPowerService.getConceptByUri(sourceURI);
+
+            if (conceptCache != null) {
+                node.setLabel(conceptCache.getWord());
+                node.setDescription(conceptCache.getDescription());
+                node.setAlternativeUris(conceptCache.getAlternativeUris());
+            }
+        } else {
+            // Need to get data from viaf if viaf URL is present, but doesn't seem to be part of story Q20-19
+            node.setLabel(event.getTerm().getPrintedRepresentation().getTermParts().iterator().next().getExpression());
+        }
+
+        return node;
+    }
+    
+    private GraphNodeData createNode(TripleElement tripleElement, GraphNodeType graphNodeType,
+            Map<String, GraphNodeData> conceptNodeMap, List<GraphData> nodes) {
+        
+        String elementUri = tripleElement.getUri();
+        //Avoid node duplication if the element is not a predicate element and if it already exists
+        if (graphNodeType != GraphNodeType.PREDICATE && elementUri != null && !elementUri.isEmpty()
+                && conceptNodeMap.containsKey(elementUri)) {
+            return conceptNodeMap.get(elementUri);
+        }
+        GraphNodeData nodeData = new GraphNodeData();
+        nodeData.setId(new ObjectId().toString());
+        nodeData.setLabel(tripleElement.getLabel());
+        nodeData.setGroup(graphNodeType.getGroupId());
+        nodeData.setUri(tripleElement.getUri());
+        if (graphNodeType != GraphNodeType.PREDICATE && elementUri != null && !elementUri.isEmpty()) {
+            conceptNodeMap.put(elementUri, nodeData);
+        }
+        nodes.add(nodeData);
+        return nodeData;
+    }
+    
+    @Override
+    public void createEdge(List<GraphData> graphEdges, String sourceId, String targetId, String eventGraphId) {
+        
+        GraphEdgeData edge = new GraphEdgeData();
+        edge.setId(new ObjectId().toString());
+        edge.setSource(sourceId);
+        edge.setTarget(targetId);
+        edge.setEventGraphId(eventGraphId);
+        graphEdges.add(edge);
+    }
+    private void createEdge(GraphData source, GraphData target, List<GraphData> edges) {
+        
+        GraphEdgeData edgeData = new GraphEdgeData();
+        edgeData.setId(new ObjectId().toString());
+        edgeData.setSource(source.getId());
+        edgeData.setTarget(target.getId());
+        edges.add(edgeData);
+    }
+    
+    private List<GraphElement> wrapInGraphElements(List<GraphData> dataList) {
+        
+        List<GraphElement> elements = new ArrayList<>();
+        dataList.forEach(data -> {
+            GraphElement element = new GraphElement();
+            element.setData(data);
+            elements.add(element);
+        });
+        return elements;
+    }
+    
+    @Override 
+    public GraphElements mapToGraph(List<DefaultMapping> triples) {
```

Comment:
missing javadoc

Reply:
Javadoc in the Interface GraphCreationService.